app/commands/v2/unpublish.rb

```diff
@@ -126,7 +126,8 @@ def find_unpublishable_content_item(content_id)
         end
 
         filter = ContentItemFilter.new(scope: ContentItem.where(content_id: content_id))
-        filter.filter(locale: locale, state: allowed_states).last
+        content_item = filter.filter(locale: locale, state: allowed_states).last
+        content_item if content_item && (payload[:allow_draft] || !Unpublishing.is_substitute?(content_item))
```

Comment:
It is a bit ugly - it's just to shortcut the need to run that query if we have a draft (as we'll only have drafts if `payload[:allow_draft]` is true